tests/integrations/server/kv_service.rs

```diff
@@ -597,13 +597,66 @@ fn test_mvcc_resolve_lock_gc_and_delete() {
     assert!(del_resp.error.is_empty());
 }
 
+#[test]
+#[cfg(feature = "failpoints")]
+fn test_mvcc_flashback_failed_in_first_batch() {
+    let (_cluster, client, ctx) = must_new_cluster_and_kv_client();
+    let mut ts = 0;
+    for i in 0..2000 {
+        let v = format!("value@{}", i).into_bytes();
+        let k = format!("key@{}", i % 1000).into_bytes();
+        // Prewrite
+        ts += 1;
+        let prewrite_start_version = ts;
+        let mut mutation = Mutation::default();
+        mutation.set_op(Op::Put);
+        mutation.set_key(k.clone());
+        mutation.set_value(v.clone());
+        must_kv_prewrite(
+            &client,
+            ctx.clone(),
+            vec![mutation],
+            k.clone(),
+            prewrite_start_version,
+        );
+        // Commit
+        ts += 1;
+        let commit_version = ts;
+        must_kv_commit(
+            &client,
+            ctx.clone(),
+            vec![k.clone()],
+            prewrite_start_version,
+            commit_version,
+            commit_version,
+        );
+        // Get
+        ts += 1;
+        must_kv_read_equal(&client, ctx.clone(), k.clone(), v.clone(), ts)
+    }
+    // Flashback
+    fail::cfg("flashback_failed_in_first_batch", "return").unwrap();
+    must_flashback_to_version(&client, ctx.clone(), 5, ts + 1, ts + 2);
+    fail::remove("flashback_failed_in_first_batch");
+    // Flashback needs to be continued.
+    must_flashback_to_version(&client, ctx.clone(), 5, ts + 1, ts + 2);
+    ts += 2;
+    // Subsequent batches of writes are deleted.
+    assert_eq!(
+        kv_read(&client, ctx, b"key@500".to_vec(), ts).value,
+        b"".to_vec()
+    );
+}
+
 #[test]
 fn test_mvcc_flashback() {
     let (_cluster, client, ctx) = must_new_cluster_and_kv_client();
     let mut ts = 0;
-    let k = b"key".to_vec();
-    for i in 0..10 {
+    let k = b"key@1".to_vec();
```

Comment:
```suggestion
```

Reply:
This value is needed later in the code and I will change its position.